qiskit_experiments/test/mock_iq_helpers.py

```diff
@@ -89,13 +91,233 @@ def compute_probabilities(self, circuits: List[QuantumCircuit])
+            prob_help_list[idx] = {
+                "physical_qubits": experiment.physical_qubits,
+                "prob": exp_helper.compute_probabilities(experiment_circuits),
+                "phase": exp_helper.iq_phase(experiment_circuits),
+                "num_circuits": len(experiment_circuits),
+            }
+
+        return prob_help_list
+
+    def _verify_parameters(
+        self,
+        exp_list: List[BaseExperiment] = None,
+        exp_helper_list: List[MockIQExperimentHelper] = None,
+    ):
+        """Check parameters before computing probability"""
+        if exp_helper_list is None:
+            raise ValueError("Please set the experiment helper list.")
+        if exp_list is None:
+            raise ValueError("Please set the experiment list.")
+
+        number_of_experiments = len(exp_list)
+        number_of_helpers = len(exp_helper_list)
+
+        if number_of_experiments == 0:
+            raise ValueError("The experiment list cannot be empty.")
+        if number_of_helpers == 0:
+            raise ValueError("The experiment helper list cannot be empty.")
+
+        if number_of_experiments != number_of_helpers:
+            raise ValueError(
+                "The number of helpers {} and the number of experiment {} don't match.".format(
+                    number_of_experiments, number_of_helpers
+                )
+            )
+
+        for helper in exp_helper_list:
+            # checking there is no nested parallel experiment.
+            if isinstance(helper, MockIQParallelExperimentHelper):
+                raise QiskitError("Nested parallel experiments aren't currently supported.")
+
+    def _parallel_exp_circ_splitter(self, qc_list: List[QuantumCircuit]):
+        """
+        Splits quantum circuits to their parallel components.
+        Args:
+            qc_list: The list of quantum circuits the backend gets as input.
+
+        Returns:
+            List: A list for each experiment. Each entry is a list of quantum circuits corresponding to
+            that experiment.
+
+        Raises:
+            QiskitError: If an instruction is applied with qubits that don't belong to the same
+            experiment.
+        """
+        # exp_idx_map connects an experiment to its circuit in the output.
+        exp_idx_map = {exp: exp_idx for exp_idx, exp in enumerate(self.exp_list)}
+        qubit_exp_map = self._create_qubit_exp_map()
+
+        exp_circuits_list = [[] for _ in self.exp_list]
+
+        for qc in qc_list:
+            # Quantum Register to qubit mapping
+            qubit_indices = {bit: idx for idx, bit in enumerate(qc.qubits)}
+
+            # initialize quantum circuit for each experiment for this instance of circuit to fill
+            # with instructions.
+            for exp_circuit in exp_circuits_list:
+                # we copy the circuit to ensure that the circuit properties (e.g. calibration and qubits
+                # frequencies) are the same in the new circuit.
```

Comment:
https://github.com/Qiskit/qiskit-terra/pull/8134, when ready, may be relevant here